coalib/misc/DictUtilities.py

```diff
@@ -11,26 +11,31 @@ def inverse_dicts(*dicts):
     No order is preserved.
 
     :param dicts: The dictionaries to invert.
+    :type: dict
     :return:      The inversed dictionary which merges all dictionaries into
                   one.
+    :rtype: defaultdict of lists
     """
-    inverse = {}
+    inverse = defaultdict(list)
 
     for dictionary in dicts:
         for key, value in dictionary.items():
             if isinstance(value, Iterable):
                 for item in value:
-                    add_pair_to_dict(item, key, inverse)
+                    inverse[item].append(key)
             else:
-                add_pair_to_dict(value, key, inverse)
-
+                inverse[value].append(key)
     return inverse
 
 
 def add_pair_to_dict(key, value, dictionary):
```

Comment:
this function is not marked as private but I don't see anyone externally seriously importing stuff from coalib.misc.DictUtilities, and it's not in bears. I'm voting for removal definitely although idealisticly you would have to deprecate but realistically I'm super sure nobody is and would not want to maintain this fuction further.

If we deprecate it it should be decorated or so by a deprecation decorator.